src/openforms/js/components/admin/form_design/logic/actions/dmn/DMNActionConfig.js

```diff
@@ -0,0 +1,232 @@
+import {Form, Formik, useFormikContext} from 'formik';
+import PropTypes from 'prop-types';
+import React, {useContext, useState} from 'react';
+import {FormattedMessage, defineMessage} from 'react-intl';
+import {useAsync} from 'react-use';
+
+import {FormContext} from 'components/admin/form_design/Context';
+import {
+  DMN_DECISION_DEFINITIONS_LIST,
+  DMN_DECISION_DEFINITIONS_VERSIONS_LIST,
+} from 'components/admin/form_design/constants';
+import Field from 'components/admin/forms/Field';
+import Select from 'components/admin/forms/Select';
+import {get} from 'utils/fetch';
+
+import VariableMapping from './VariableMapping';
+import {inputValuesType} from './types';
+
+const ERRORS = {
+  required: defineMessage({
+    description: 'Required error message',
+    defaultMessage: 'This field is required.',
+  }),
+};
+
+const DecisionDefinitionIdField = () => {
+  const {values, setFieldValue, getFieldProps, errors, touched} = useFormikContext();
+  const [decisionDefinitions, setDecisionDefinitions] = useState([]);
+
+  useAsync(async () => {
+    if (!values.pluginId) {
+      setFieldValue('decisionDefinitionId', '');
+      setDecisionDefinitions([]);
+      return;
+    }
+
+    const response = await get(DMN_DECISION_DEFINITIONS_LIST, {
+      engine: values.pluginId,
+    });
+
+    const definitionChoices = response.data.map(item => [item.id, item.label]);
+    setDecisionDefinitions(definitionChoices);
+
+    // If pluginId has changed, reset the value of the definition
+    if (
+      values.decisionDefinitionId !== '' &&
+      !definitionChoices.find(item => item[0] === values.decisionDefinitionId)
+    ) {
+      setFieldValue('decisionDefinitionId', '');
+    }
+  }, [values.pluginId, setFieldValue]);
+
+  return (
+    <Field
+      name="decisionDefinitionId"
+      htmlFor="decisionDefinitionId"
+      label={
+        <FormattedMessage
+          defaultMessage="Decision definition ID"
+          description="Decision definition ID label"
+        />
+      }
+      errors={
+        touched.decisionDefinitionId && errors.decisionDefinitionId
+          ? [ERRORS[errors.decisionDefinitionId]]
+          : []
+      }
+    >
+      <Select
+        id="decisionDefinitionId"
+        name="decisionDefinitionId"
+        allowBlank={true}
+        choices={decisionDefinitions}
+        {...getFieldProps('decisionDefinitionId')}
+      />
+    </Field>
+  );
+};
+
+const DecisionDefinitionVersionField = () => {
+  const {
+    values: {pluginId, decisionDefinitionId},
+    setFieldValue,
+    getFieldProps,
+  } = useFormikContext();
```

Comment:
with the low-level usage, perhaps using `useField(...)` is more appropriate?

You'll also need `useFormikContext` to get access to the other field values, but `useField` has the advantage that is also registers the field with the formik state.